cloud/scope/cluster_test.go

```diff
@@ -97,6 +97,114 @@ func TestClusterScope_ReconcileFailureDomains(t *testing.T) {
+		CompartmentId:      common.String("3ad"),
+		AvailabilityDomain: common.String("ad2"),
+	})).Return(identity.ListFaultDomainsResponse{Items: []identity.FaultDomain{
+		{
+			Name:               common.String("fault-domain-1"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fault-domain-2"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fault-domain-3"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+	}}, nil)
+
+	identityClient.EXPECT().ListFaultDomains(gomock.Any(), gomock.Eq(identity.ListFaultDomainsRequest{
+		CompartmentId:      common.String("3ad"),
+		AvailabilityDomain: common.String("ad3"),
+	})).Return(identity.ListFaultDomainsResponse{Items: []identity.FaultDomain{
+		{
+			Name:               common.String("fault-domain-1"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fault-domain-2"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fault-domain-3"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+	}}, nil)
+
+	identityClient.EXPECT().ListFaultDomains(gomock.Any(), gomock.Eq(identity.ListFaultDomainsRequest{
+		CompartmentId:      common.String("1ad"),
+		AvailabilityDomain: common.String("ad1"),
+	})).Return(identity.ListFaultDomainsResponse{Items: []identity.FaultDomain{
+		{
+			Name:               common.String("fd1"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fd2"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fd3"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+	}}, nil)
+
+	identityClient.EXPECT().ListFaultDomains(gomock.Any(), gomock.Eq(identity.ListFaultDomainsRequest{
+		CompartmentId:      common.String("2ad"),
+		AvailabilityDomain: common.String("ad1"),
+	})).Return(identity.ListFaultDomainsResponse{Items: []identity.FaultDomain{
+		{
+			Name:               common.String("fd1"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fd2"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fd3"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+	}}, nil)
+
+	identityClient.EXPECT().ListFaultDomains(gomock.Any(), gomock.Eq(identity.ListFaultDomainsRequest{
+		CompartmentId:      common.String("2ad"),
+		AvailabilityDomain: common.String("ad2"),
+	})).Return(identity.ListFaultDomainsResponse{Items: []identity.FaultDomain{
+		{
+			Name:               common.String("fd1"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fd2"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+		{
+			Name:               common.String("fd3"),
+			AvailabilityDomain: common.String("ad1"),
+		},
+	}}, nil)
+
```

Comment:
Since a new `IdentityClient.ListFaultDomains` was added in the `setFailureDomains` call in clients.go these new tests ssserts needed to be added.